web_sdk/test/api_conform_test.dart

```diff
@@ -39,10 +39,70 @@ void main() {
     }
     // Next will check that the public methods exposed in each library are
     // identical.
-    final Map<String, MethodDeclaration> uiMethods = <String, MethodDeclaration>{};
-    final Map<String, MethodDeclaration> webMethods = <String, MethodDeclaration>{};
+    final Map<String, MethodDeclaration> uiMethods =
+        <String, MethodDeclaration>{};
+    final Map<String, MethodDeclaration> webMethods =
+        <String, MethodDeclaration>{};
+    final Map<String, ConstructorDeclaration> uiConstructors =
+        <String, ConstructorDeclaration>{};
+    final Map<String, ConstructorDeclaration> webConstructors =
+        <String, ConstructorDeclaration>{};
     _collectPublicMethods(uiClass, uiMethods);
     _collectPublicMethods(webClass, webMethods);
+    _collectPublicConstructors(uiClass, uiConstructors);
+    _collectPublicConstructors(webClass, webConstructors);
+
+    for (String name in uiConstructors.keys) {
+      final ConstructorDeclaration uiConstructor = uiConstructors[name];
+      final ConstructorDeclaration webConstructor = webConstructors[name];
+      if (webConstructor == null) {
+        failed = true;
+        print(
+          'Warning: lib/ui/ui.dart $className.$name is missing from lib/stub_ui/ui.dart.',
+        );
+        continue;
+      }
+
+      if (uiConstructor.parameters.parameters.length !=
+          webConstructor.parameters.parameters.length) {
+        failed = true;
+        print(
+            'Warning: lib/ui/ui.dart $className.$name has a different parameter '
+            'length than in lib/stub_ui/ui.dart.');
+      }
+
+      for (int i = 0;
+          i < uiConstructor.parameters.parameters.length &&
+              i < uiConstructor.parameters.parameters.length;
+          i++) {
+        // Technically you could re-order named parameters and still be valid,
+        // but we enforce that they are identical.
+        for (int i = 0;
+            i < uiConstructor.parameters.parameters.length &&
+                i < webConstructor.parameters.parameters.length;
+            i++) {
+          final FormalParameter uiParam =
+              uiConstructor.parameters.parameters[i];
+          final FormalParameter webParam =
+              webConstructor.parameters.parameters[i];
+          if (webParam.identifier.name != uiParam.identifier.name) {
+            failed = true;
+            print('Warning: lib/ui/ui.dart $className.$name parameter $i'
+                ' ${uiParam.identifier.name} has a different name in lib/stub_ui/ui.dart.');
+          }
+          if (uiParam.isPositional && !webParam.isPositional) {
```

Comment:
Should this be `uiParam.isPositional != webParam.isPositional`?

(same for `isNamed`)

Reply:
Fixed